engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/ExecutionStore.scala

```diff
@@ -19,44 +19,76 @@ object ExecutionStore {
       case scatter: Scatter => Option(ScatterKey(scatter))
       case conditional: If => Option(ConditionalKey(conditional, None))
       case declaration: Declaration => Option(DeclarationKey(declaration, None, workflowCoercedInputs))
-      case _ => None // Ifs will need to be added here when supported
+      case _ => None
     }
 
     new ExecutionStore(keys.flatten.map(_ -> NotStarted).toMap)
   }
 }
 
-case class ExecutionStore(store: Map[JobKey, ExecutionStatus]) {
+case class ExecutionStore(private val store: Map[JobKey, ExecutionStatus]) {
 
-  override def toString = store.map { case (j, s) => s"$j -> $s" } mkString(System.lineSeparator())
+  // This is hugely ripe for optimization, if it becomes a bottleneck
```

Comment:
I'm glad that you're going to officially take ownership of this comment in the git blame!